### ISSUE TITLE
HasWidgetsOverlay causes exception after restart/reload

### ISSUE THREAD

**Issue by abegehr** [flame-engine/flame#236]
calling addWidgetOverlay or removeWidgetOverlay after restarting/reloading the app in debug mode (on iOS Simulator), causes an exception.

The exception:
```
════════ Exception caught by gesture ═══════════════════════════════════════════
The following StateError was thrown while handling a gesture:
**Bad state: Cannot add event after closing**

When the exception was thrown, this was the stack
#2      HasWidgetsOverlay.removeWidgetOverlay 
package:flame/game/game.dart:94
#3      Jump.start 
package:jump/game.dart:40
#4      _InkResponseState._handleTap 
package:flutter/…/material/ink_well.dart:705
#5      _InkResponseState.build.<anonymous closure> 
package:flutter/…/material/ink_well.dart:788
#6      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:182
...
Handler: "onTap"
Recognizer: TapGestureRecognizer#0ae50
    debugOwner: GestureDetector
    state: possible
    won arena
    finalPosition: Offset(191.0, 406.0)
    finalLocalPosition: Offset(58.0, 9.5)
    button: 1
    sent tap down
════════════════════════════════════════════════════════════════════════════════
```

Cause:

I think this is, because HasWidgetsOverlay's StreamController `widgetOverlayController` is closed in the Game widget's `onDetach()` method: https://github.com/flame-engine/flame/blob/master/lib/game/game.dart#L70.

Logging the widgetOverlayController.isClosed after reload/restart prints 'true'. This is a hassle since the app would need to rebuild on each code change when using HasWidgetsOverlay.

Possible fix:

Not closing the StreamController on 'game.onDispatch()' solves the issue for restarting and reloading in debug mode. This is related: https://github.com/flutter/flutter/issues/16426

Only closing the `widgetOverlayController` StreamController on `game.onDetach()` in release mode could be a solution:
```
if (this is HasWidgetsOverlay && kReleaseMode) {
    (this as HasWidgetsOverlay).widgetOverlayController.close();
}
```
https://github.com/flame-engine/flame/blob/master/lib/game/game.dart#L70
– @erickzanardo, is this a viable solution?

**Comment by erickzanardo** [comment]
This seems indeed like a good solution, I will probably try that, but I would also like to hear @luanpotter and @renancaraujo thoughts about this.

**Comment by erickzanardo** [comment]
fixed on #241
